engine/apps/mobile_app/views.py

```diff
@@ -72,3 +87,110 @@ def notification_timing_options(self, request):
             {"value": item[0], "display_name": item[1]} for item in MobileAppUserSettings.NOTIFICATION_TIMING_CHOICES
         ]
         return Response(choices)
+
+
+class MobileAppGatewayView(APIView):
+    authentication_classes = (MobileAppAuthTokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    class SupportedDownstreamBackends:
+        INCIDENT = "incident"
+
+    ALL_SUPPORTED_DOWNSTREAM_BACKENDS = [
+        SupportedDownstreamBackends.INCIDENT,
+    ]
+
+    @classmethod
+    def _construct_jwt_payload(cls, user: "User") -> typing.Dict[str, typing.Any]:
+        organization = user.organization
+        now = timezone.now()
+
+        return {
+            # registered claim names
+            "iat": now,
+            "exp": now + timezone.timedelta(minutes=1),  # jwt is short lived. expires in 1 minute.
+            # custom data
+            "user_id": user.user_id,  # grafana user ID
+            "stack_id": organization.stack_id,
+            "organization_id": organization.org_id,  # grafana org ID
+            "stack_slug": organization.stack_slug,
+            "org_slug": organization.org_slug,
+        }
+
+    @classmethod
+    def _construct_jwt(cls, user: "User") -> str:
+        """
+        RS256 = asymmetric = public/private key pair
+        HS256 = symmetric = shared secret (don't use this)
+        """
+        return jwt.encode(
+            cls._construct_jwt_payload(user), settings.MOBILE_APP_GATEWAY_RSA_PRIVATE_KEY, algorithm="RS256"
+        )
+
+    @classmethod
+    def _get_downstream_headers(cls, user: "User") -> typing.Dict[str, str]:
+        return {
+            "Authorization": f"Bearer {cls._construct_jwt(user)}",
+        }
+
+    @classmethod
+    def _get_downstream_url(cls, organization: "Organization", downstream_backend: str, downstream_path: str) -> str:
+        downstream_url = {
+            cls.SupportedDownstreamBackends.INCIDENT: organization.grafana_incident_backend_url,
```

Comment:
in the future, if we want to add a new proxied backend, we simply need to add a mapping here from the URLConf's `downstream_backend` to the downstream service's root URL